src/audio/component.c

```diff
@@ -506,6 +516,54 @@ int comp_copy(struct comp_dev *dev)
 	return ret;
 }
 
+#ifdef CONFIG_SOF_TELEMETRY_PERFORMANCE_MEASUREMENTS
+void comp_init_performance_data(struct comp_dev *dev)
+{
+	struct perf_data_item_comp *item = dev->perf_data.perf_data_item;
+
+	if (item)
+		perf_data_item_comp_init(item, dev->ipc_config.id, 0);
+}
+
+/* returns true if budget violation occurred */
+static bool update_peak_of_measured_cpc(struct comp_dev *dev, size_t measured_cpc)
+{
+	if (measured_cpc <= dev->perf_data.peak_of_measured_cpc)
+		return false;
+	dev->perf_data.peak_of_measured_cpc = measured_cpc;
+	return measured_cpc > dev->cpc;
+}
+
+bool comp_update_performance_data(struct comp_dev *dev, uint32_t cycles_used)
+{
+	struct perf_data_item_comp *item = dev->perf_data.perf_data_item;
+
+	if (perf_meas_get_state() == IPC4_PERF_MEASUREMENTS_STARTED) {
+		/* we divide by ibs so we need to check if its set */
+		if (item && dev->ibs != 0) {
+			item->total_iteration_count++;
+			if (item->total_iteration_count == 0) {
+				/* We can't allow count to overflow to 0. Overflow will also make
+				 * the results incorrect. We don't want to crash in this case,
+				 * so we just log it.
+				 */
+				item->total_iteration_count = 1;
+				tr_err(&ipc_tr,
+				       "overflow for module %#x, performance measurement incorrect",
+				       dev_comp_id(dev));
```

Comment:
but like this after such an overflow they'll stay incorrect? Can we reset `total_cycles_consumed` to start producing valid data again?

Reply:
Hm, we can either do that, or zero everything and set a flag to stop the measurement to further indicate that something went wrong.